server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -1477,6 +1478,16 @@ private Pair<List<Long>, Integer> searchForUserVMIdsAndCount(ListVMsCmd cmd) {
             userVmSearchBuilder.join("tags", resourceTagSearch, resourceTagSearch.entity().getResourceId(), userVmSearchBuilder.entity().getId(), JoinBuilder.JoinType.INNER);
         }
 
+        if (cmd.getOnlyLeasedInstances() != null && cmd.getOnlyLeasedInstances()) {
+            if (VMLeaseManagerImpl.InstanceLeaseEnabled.value()) {
+                SearchBuilder<UserVmDetailVO> leasedInstancesSearch = userVmDetailsDao.createSearchBuilder();
+                leasedInstancesSearch.and(leasedInstancesSearch.entity().getName(), SearchCriteria.Op.EQ).values(VmDetailConstants.INSTANCE_LEASE_EXPIRY_DATE);
+                userVmSearchBuilder.join("userVmToLeased", leasedInstancesSearch, leasedInstancesSearch.entity().getResourceId(), userVmSearchBuilder.entity().getId(), JoinBuilder.JoinType.INNER);
+            } else {
+                logger.warn("Lease feature is not enabled, onlyleasedinstances will be considered as false");
```

Comment:
When _onlyleasedinstances_ is passed as true, better communicate this msg to the user.